tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp

```diff
@@ -393,27 +402,32 @@ convertFileContentsToInputs(const SmallVectorImpl<FileContent> &contents) {
   return inputsAndOutputs;
 }
 
-static FrontendInputsAndOutputs
-resolveSymbolicLinksInInputs(FrontendInputsAndOutputs &inputsAndOutputs,
-                             StringRef UnresolvedPrimaryFile,
-                             std::string &Error) {
+static FrontendInputsAndOutputs resolveSymbolicLinksInInputs(
+    FrontendInputsAndOutputs &inputsAndOutputs, StringRef UnresolvedPrimaryFile,
+    std::string &Error,
+    llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FileSystem) {
+  if (!FileSystem)
+    FileSystem = llvm::vfs::getRealFileSystem();
```

Comment:
Do you think it would be useful to require that `FileSystem` parameters are always non-null, so that it's the client's responsibility to give us `llvm::vfs::getRealFileSystem()` (or that could be turned into a default value)? It looks like that would avoid quite a few of checks like that. Alternatively, maybe `FileSystem` parameters could have a non-pointer type that encapsulates a pointer and falls back to the real file system if the pointer is null?

Reply:
will be addressed by the small refactoring